scripting/include/srccoop/manager.inc

```diff
@@ -107,12 +138,44 @@ methodmap CoopManager
 		delete data.kv;
 	}
 
-	public static void OnClientPutInServer(CBasePlayer pPlayer)
+	public static void SetIntroTimeHud(const int iSecondsToWait)
 	{
-		if (CoopManager.IsCoopModeEnabled())
+		bool bShouldShow = iSecondsToWait > -1;
+		
+		#if defined SRCCOOP_BLACKMESA
+		CBM_MP_GameRules.SetStateEndTime(STATE_WARMUP, bShouldShow ? (GetGameTime() + iSecondsToWait + 1.0) : 0.0);
+		#endif
+
+		if (bShouldShow)
 		{
-			CoopManager.CheckStart();
-			CCoopSpawnSystem.TeleportPlayerOnSpawn(pPlayer);
+			#if defined INTRO_TIMER_HUDTEXT
+			SetHudTextParams(-1.0, 0.05, 1.0, 200, 200, 200, 255, 0, 0.5, 0.1, 0.1);
+			for (int i = 1; i <= MaxClients; i++)
+			{
+				if (IsClientInGame(i))
+				{
+					ShowHudText(i, 2, "%d", data.m_iSecondsToStart);
+				}
+			}
+			#endif
+			
+			if (data.m_iSecondsToStart > 0)
+			{
+				EmitSoundToAll(Conf.SND_TICK, .volume = Conf.SND_TICK_VOL);
+			}
+
+			if (data.m_iSecondsToStart == 3)
+			{
+				EmitSoundToAll(Conf.SND_3, .volume = Conf.SND_TICK_VOL);
+			}
+			else if (data.m_iSecondsToStart == 2)
+			{
+				EmitSoundToAll(Conf.SND_2, .volume = Conf.SND_TICK_VOL);
+			}
+			else if (data.m_iSecondsToStart == 1)
+			{
+				EmitSoundToAll(Conf.SND_1, .volume = Conf.SND_TICK_VOL);
+			}
 		}
 	}
```

Comment:
Some parts refer to iSecondsToWait, other to data.m_iSecondsToStart

Reply:
Fixed this to entirely use `iSecondsToWait`.